packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoHandlingDelegate.kt

```diff
@@ -99,11 +99,20 @@ class ExpoHandlingDelegate(protected val context: Context) : HandlingDelegate {
       getListeners().forEach {
         it.onNotificationReceived(notification)
       }
-    } else {
+    } else if (notification.shouldPresent()) {
       NotificationsService.present(context, notification)
     }
   }
 
+  /**
+   * If the app is backgrounded, a notification is only presented if
+   * the title and text are both non-empty. Otherwise, this is a "data-only" or "silent"
+   * notification that should not be presented to the user.
+   */
+  private fun Notification.shouldPresent(): Boolean {
+    return !(notificationRequest.content.title.isNullOrEmpty() && notificationRequest.content.text.isNullOrEmpty())
```

Comment:
i was wondering for the logic here. if title is empty but text is not empty, do we present the notification? from the implementation and test code i think the answer is yes, but from the comment above, i think it's no.

> a notification is only presented if the title and text are both non-empty.

could you help me to clarify that?

Reply:
Good catch! This comment was in error. If there is a title only or text only, notifications should still be displayed. If both are missing then this is a "silent notification".

I've updated the comment to reflect this 😄.